pkg/volume/secret/secret.go

```diff
@@ -126,7 +127,7 @@ func (sv *secretVolume) SetUpAt(dir string) error {
 	for name, data := range secret.Data {
 		hostFilePath := path.Join(dir, name)
 		glog.V(3).Infof("Writing secret data %v/%v/%v (%v bytes) to host file %v", sv.podRef.Namespace, sv.secretName, len(data), hostFilePath)
-		err := ioutil.WriteFile(hostFilePath, data, 0777)
+		err := ioutil.WriteFile(hostFilePath, data, 0555)
```

Comment:
@thockin I noticed that we were setting the perms foolishly -- this seems like a better mode.